crates/ruff_db/src/vfs.rs

```diff
@@ -315,7 +324,7 @@ impl VendoredVfs {
 
     fn read(&self, path: &VendoredPath) -> Option<String> {
         match self {
-            VendoredVfs::Real => todo!(),
+            VendoredVfs::Real(file_system) => file_system.read(path),
```

Comment:
Nit: Maybe rename to `vendored` to avoid confusion with `FileSystem`
```suggestion
            VendoredVfs::Real(vendored) => vendored.read(path),
```

Reply:
It _is_ a file system though, it just isn't an _OS_ file system :) that's what the `fs` in `Vfs` stands for, right?

I don't think `vendored` as a name works well here, because the distinction between the two enum branches is whether it's a "Real file system" or a "Stubbed file system", not whether it's a "Real vendored" or a "Stubbed vendored" (that doesn't really make sense).

I would prefer it if we considered renaming the `FileSystem` trait instead

Reply:
Yeah not sure. The `FileSystem` maps to `std::fs` and vendored is not that. I don't think `FileSystem` maps to the `OS` file system, because an in memory (or wasm FS) isn't the OS's file system.

I'm currently thinking about renaming `VfsFile` to `Files`, `VfsFile` to `File` (not sure about `VfsPath`). `VendoredFiles` would then fit well into this naming and would also make it clear, that it only supports files. But I'm okay deferring this for now.